### PR TITLE
add README and auth values from env vars

### PR DESCRIPTION
**Pull request by uwedeportivo**
Let's you specify the OAuth2 tokens in env vars to avoid having them in repos.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Terraform Provider for Site24x7
+
+This is a [custom provider]( https://www.terraform.io/docs/extend/writing-custom-providers.html) 
+for [Site24x7 API](https://www.site24x7.com/help/api/) 
+
+`example.tf` file with the provider section:
+
+```
```
Comment (keegancsmith) [inline README.md:8]:
```suggestion
```terraform
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+You can choose to specify the OAuth2 values as environment variables instead of including them in the provider section:
+
+* `SITE24X7_CLIENT_ID`
+* `SITE24X7_CLIENT_SECRET`
+* `SITE24X7_REFRESH_TOKEN`
+
+## Installation
+
+```code
```
Comment (keegancsmith) [inline README.md]:
```suggestion
```shellsession
```

**Review by keegancsmith — APPROVED** [review]

**Comment by nickcostigan** [comment]
I tried this branch out and was able to successfully read the client secret from an environment variable

**Comment by uwedeportivo** [comment]
oops, forgot to commit :-), sorry

**Pull request merged**